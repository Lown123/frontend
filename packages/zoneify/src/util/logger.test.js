/* eslint-disable no-console */
const {
  formatter, log, levels, spinner,
} = require('./logger')

global.console = {
  log: jest.fn(),
}

describe('logger', () => {
  const message = 'test'
  const debugOutput = formatter({ level: 'debug', message })
  const errorOutput = formatter({ level: 'error', message })
  const infoOutput = formatter({ level: 'info', message })

  it('should have log levels', () => {
    expect(levels).toMatchSnapshot()
  })

  it('should a default log level of info', () => {
    expect(log.level).toBe('info')
  })

  Object.keys(levels).forEach((level) => {
    it(`should have a method for ${level}`, () => {
      expect(typeof log[level]).toBe('function')
    })

    it(`should set the log level to ${level}`, () => {
      log.level = level
      expect(log.level).toBe(level)
    })
  })

  it('should set not change the log level', () => {
    const originalLevel = log.level

    log.level = 'hello world'
    expect(log.level).toBe(originalLevel)
  })

  it('should format debug logs differently to info logs', () => {
    expect(debugOutput).not.toEqual(infoOutput)
  })

  it('should format error logs differently to info logs', () => {
    expect(errorOutput).not.toEqual(infoOutput)
  })

  it('should format error logs differently to debug logs', () => {
    expect(errorOutput).not.toEqual(infoOutput)
  })

  it('should start/stop the spinner', () => {
    const spinnerMessage = 'Test'
    const start = jest.spyOn(spinner, 'start')
    const stop = jest.spyOn(spinner, 'stop')

    spinner.start(spinnerMessage)
    expect(spinner.isSpinning).toBeTruthy()
    log.info('Test')
    expect(console.log).toBeCalled()
    expect(stop).toBeCalled()
    expect(start).toBeCalledWith(spinnerMessage)
    expect(spinner.isSpinning).toBeTruthy()
    expect(spinner.text).toBe(spinnerMessage)
    spinner.stop()
    start.mockReset()
    stop.mockReset()
  })

  it('should not start/stop the spinner', () => {
    const start = jest.spyOn(spinner, 'start')
    const stop = jest.spyOn(spinner, 'stop')

    expect(spinner.isSpinning).toBeFalsy()
    log.info('Test')
    expect(console.log).toBeCalled()
    expect(stop).not.toBeCalled()
    expect(start).not.toBeCalled()
    expect(spinner.isSpinning).toBeFalsy()
  })

  it('should not log debug messages when info', () => {
    log.level = 'info'
    expect(log.level).toBe('info')
    log.debug('Test debug')
    expect(console.log).not.toBeCalled()
  })

  afterEach(() => {
    console.log.mockReset()
  })
})
