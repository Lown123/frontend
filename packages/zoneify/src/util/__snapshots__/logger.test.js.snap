// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logger should format debug logs 1`] = `"[2m[1m[2mDEBUG[1m[2m [2mtest[2m[22m"`;

exports[`logger should format error logs 1`] = `"[31m[1mERROR[22m test[39m"`;

exports[`logger should format info logs 1`] = `"test"`;

exports[`logger should have log levels 1`] = `
Object {
  "debug": 2,
  "error": 0,
  "info": 1,
}
`;
